test/functional/terminal/tui_spec.lua

```diff
@@ -871,14 +871,22 @@ describe('TUI background color', function()
     screen:expect{any='did OptionSet, yay!'}
   end)
 
+  local last_bg
   local function assert_bg(colorspace, color, bg)
     it('handles '..color..' as '..bg, function()
+
+      -- Ensure the opposite of the expected bg is active.
+      if bg == last_bg then
+        command('set bg=' .. (bg == 'dark' and 'light' or 'dark'))
+      end
+      last_bg = bg
+
       feed_data('\027]11;'..colorspace..':'..color..'\007')
       -- Retry until the terminal response is handled.
       retry(100, nil, function()
         feed_data(':echo &background\n')
         screen:expect({
-          timeout=40,
+          timeout=20,
```

Comment:
Why does this need `screen:expect` in the first place?
I.e. what is missing with just `eq('&bg', bg')`?  Could redrawing the screen be triggered explicitly somehow?
(not wanting to change this necessarily, but just out of curiosity)

Reply:
because `eval('&bg')` (assuming that is what you meant) will be executed in the terminal host instance, not the instance running inside the terminal.

Reply:
Yes, I've meant `eval('&bg')`.

> will be executed in the terminal host instance, not the instance running inside the terminal.

This test is not about `:terminal`, is it?
Isn't `command()` triggering a command in the Nvim instance under test?

Reply:
> This test is not about :terminal, is it?

No. Therefore we we want to inspect the state of the nvim process running the TUI, not the one running `:terminal`, which was what I said. 

> Isn't command() triggering a command in the Nvim instance under test?

Unfortunately not in the `terminal/tui_spec.lua` module, we need to use `:terminal` to "drive" the TUI tests. Maybe the comment in the beginning of this file could be expanded.